plugins/processors/awsappsignals/processor.go

```diff
@@ -0,0 +1,268 @@
+
+	ap.customReplacer = customconfiguration.NewCustomReplacer(ap.config.Rules)
+	ap.traceMutators = append(ap.traceMutators, attributesResolver, attributesNormalizer, ap.customReplacer)
+
+	customKeeper := customconfiguration.NewCustomKeeper(ap.config.Rules)
+	ap.customAllowlistMutators = append(ap.customAllowlistMutators, customKeeper)
+	customDropper := customconfiguration.NewCustomDropper(ap.config.Rules)
+	ap.customAllowlistMutators = append(ap.customAllowlistMutators, customDropper)
+
+	return nil
+}
+
+func (ap *awsappsignalsprocessor) Shutdown(ctx context.Context) error {
+	for _, stopper := range ap.stoppers {
+		err := stopper.Stop(ctx)
+		if err != nil {
+			ap.logger.Error("failed to stop", zap.Error(err))
+		}
+	}
+	return nil
+}
+
+func (ap *awsappsignalsprocessor) processTraces(ctx context.Context, td ptrace.Traces) (ptrace.Traces, error) {
+	rss := td.ResourceSpans()
+	for i := 0; i < rss.Len(); i++ {
+		rs := rss.At(i)
+		ilss := rs.ScopeSpans()
+		resourceAttributes := rs.Resource().Attributes()
+		for j := 0; j < ilss.Len(); j++ {
+			ils := ilss.At(j)
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				for _, Mutator := range ap.traceMutators {
+					err := Mutator.Process(span.Attributes(), resourceAttributes, true)
+					if err != nil {
+						ap.logger.Debug("failed to Process span", zap.Error(err))
+					}
+				}
+			}
+		}
+	}
+	return td, nil
+}
+
+func (ap *awsappsignalsprocessor) processMetrics(ctx context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
+	rms := md.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rs := rms.At(i)
+		ilms := rs.ScopeMetrics()
+		resourceAttributes := rs.Resource().Attributes()
+		for j := 0; j < ilms.Len(); j++ {
+			ils := ilms.At(j)
+			metrics := ils.Metrics()
+			for k := 0; k < metrics.Len(); k++ {
+				m := metrics.At(k)
+				ap.processMetricAttributes(ctx, m, resourceAttributes)
+			}
+		}
+	}
+	return md, nil
+}
+
+// Attributes are provided for each log and trace, but not at the metric level
+// Need to process attributes for every data point within a metric.
+func (ap *awsappsignalsprocessor) processMetricAttributes(ctx context.Context, m pmetric.Metric, resourceAttribes pcommon.Map) {
+
+	// This is a lot of repeated code, but since there is no single parent superclass
+	// between metric data types, we can't use polymorphism.
+	switch m.Type() {
+	case pmetric.MetricTypeGauge:
+		dps := m.Gauge().DataPoints()
+		for i := 0; i < dps.Len(); i++ {
+			for _, Mutator := range ap.metricMutators {
+				err := Mutator.Process(dps.At(i).Attributes(), resourceAttribes, false)
+				if err != nil {
+					ap.logger.Debug(failedToProcessAttribute, zap.Error(err))
+				}
+			}
+		}
+		dps.RemoveIf(func(d pmetric.NumberDataPoint) bool {
+			for _, Mutator := range ap.customAllowlistMutators {
+				shouldBeDropped, err := Mutator.ShouldBeDropped(d.Attributes(), resourceAttribes)
```

Comment:
nit: misspelling of `resourceAttributes`

Reply:
fixed